### PR TITLE
Fix default num cpus

### PR DESCRIPTION
**Pull request by rc-vbundage**
This PR adds a fix to the default number of cpus when setting the `max_entries` field on the perfmap. The issue being addressed is an edge case when the number of cpus changes due to a hot swap of cpus on a system. The proposed fix is to set the `max_entries` value based on the max possible cpu count instead of currently online cpu count.

### CODE REVIEW

File: src/lib.rs [hunk 1]
```diff
@@ -831,7 +830,8 @@ impl ProgramVersion<'_> {
                     match map.definition.map_type {
                         bpf_map_type::BPF_MAP_TYPE_PERF_EVENT_ARRAY => {
                             if map.definition.max_entries == 0 {
-                                map.definition.max_entries = get_cpus()?.len() as u32;
+                                map.definition.max_entries =
+                                    cpu_info::max_possible_index()? as u32 + 1;
```
Comment (rc-andres) [inline src/lib.rs:834]: @rc-vbundage we should move this out of the for-loop so we aren't recalculating it again and again. We should have caught this, sorry!

**Review by rc-andres — APPROVED** [review]
I had some small changes we forgot to touch up on. Mainly moving the calculation outside of the for-loop. Looks good to me otherwise but we can wait for @RafaelOrtizRC 's thoughts.

File: src/cpu_info.rs [hunk 2]
```diff
@@ -0,0 +1,125 @@
...
+        |e| {
+            info!(
+                LOGGER.0,
+                "cpu_info::max_possible_index(); could not read /sys/devices/system/cpu/possible; error: {:?}", e
+            );
+            OxidebpfError::FileIOError
+        },
+    )?;
+
+    max_index(&cpu_string)
```
Comment (RafaelOrtizRC) [inline src/cpu_info.rs:28]: Anecdotally, the `online` and `possible` formats seem to be the same. In the average case, they'll match 1:1. In the edge case, it may be a superset but the highest number is not necessarily the highest count. For example, I can't see any reason `0-11,16-32` isn't a valid `possible`, where `online` might be `2-5,20-24`. Weird, but possible.

I can't find the place in the kernel where this string is written, but also looking at where CPU masks are set I don't see why this wouldn't be _possible_. Since we're already dealing with edge cases, let's just count explicitly here instead of taking the max index.

File: src/cpu_info.rs [hunk 3]
```diff
@@ -0,0 +1,125 @@
...
+
+    #[test]
+    fn single_number() {
+        let result = max_index("4").expect("did not parse cpu_string");
+        assert_eq!(result, 4);
+    }
+
+    #[test]
+    fn combination() {
+        let result = max_index("0,3-5,8\n").expect("did not parse cpu_string");
```
Comment (RafaelOrtizRC) [inline src/cpu_info.rs:115]: I don't see anywhere in the kernel where this _couldn't_ be a valid `possible` string. I also can't easily picture a scenario where that would happen (outside a sysadmin manually setting a very specific `possible` mask) but better safe than sorry.

File: src/lib.rs [hunk 4]
```diff
@@ -475,7 +474,7 @@ impl<'a> Program<'a> {
         let is_return = self.kind == Some(ProgramType::Uretprobe);
         let pid = self.pid.unwrap_or(-1);
 
-        maps::get_cpus()?
+        cpu_info::online()?
```
Comment (RafaelOrtizRC) [inline src/lib.rs:477]: Thank you for factoring this out into its own module!

**Review by RafaelOrtizRC — CHANGES_REQUESTED** [review]
I just have one minor edge case concern on top of this edge case.

Also, welcome @rc-vbundage to being an `oxidebpf` contributor! 🎉

**Review by RafaelOrtizRC — APPROVED** [review]
Clarified with @rc-andres that we're only using this to grab the index values, and we're still allocating based on `online` count.
